tests/testthat/test-schedule.R

```diff
@@ -0,0 +1,211 @@
+library(R6)
+
+MockConnect <- R6Class(
+  "MockConnect",
+  inherit = Connect,
+  public = list(
+    initialize = function(version = NA) {
+      self$server <- "https://connect.example"
+      self$api_key <- "fake"
+      private$.version <- version
+    },
+    request = function(method, url, ..., parser = "parsed") {
+      # Record call
+      self$log_call(paste(method, url))
+
+      # Look for response
+      if (!(url %in% names(self$responses))) {
+        stop("Unexpected URL")
+      }
+      res <- self$responses[[url]]
+
+      if (is.null(parser)) {
+        res
+      } else {
+        self$raise_error(res)
+        httr::content(res, as = parser)
+      }
+    },
+    responses = list(),
+    add_mock_response = function(path, content, status_code = 200L, headers = c("Content-Type" = "application/json; charset=utf-8")) {
+      url <- self$api_url(path)
+      res <- mock_response(url, content, status_code, headers)
+      self$responses[[url]] <- res
+    },
+    call_log = character(),
+    log_call = function(call) {
+      self$call_log <- c(self$call_log, call)
+    }
+  )
+)
+
+mock_response <- function(url, content, status_code, headers = character()) {
+  # Headers in responses are case-insensitive lists.
+  names(headers) <- tolower(names(headers))
+  headers <- as.list(headers)
+  headers <- structure(as.list(headers), class = c("insensitive", class(headers)))
+
+  # Treat content similarly to httr::POST, with a subset of behaviors
+  if (is.character(content) && length(content) == 1) {
+    content <- charToRaw(content)
+  } else if (is.list(content)) {
+    content <- charToRaw(toJSON(content, auto_unbox = TRUE))
+  }
+
+  structure(
+    list(
+      url = url,
+      status_code = status_code,
+      request = structure(list(url = url), class = "request"),
+      headers = headers,
+      content = content
+    ),
+    class = "response"
+  )
+}
+
+mock_response_404 <- function(url) {
+  mock_response(url, content = "404 page not found", status_code = 404L, headers = c("Content-Type" = "text/plain"))
+}
+
+
+mock_response(
+  url = "v1/timezones",
+  body = list(
+    list(timezone = "Africa/Abidjan", offset = "+00:00"),
+    list(timezone = "Africa/Accra", offset = "+00:00")
+  )
+)
```

Comment:
Is this something you're having trouble getting `httptest` to do? This pattern should be possible: 

https://github.com/Crunch-io/rcrunch/blob/b709e282172ee6b813b26299c28514ea840c47a6/tests/testthat/test-api.R#L117-L124 and the fixture https://github.com/Crunch-io/rcrunch/blob/b709e282172ee6b813b26299c28514ea840c47a6/tests/testthat/app.crunch.io/404.R#L1-L18 are one example of this.

I know that you mentioned wanting to experiment with possible other test runners / setups for the http testing, but I don't think this is the place we should do that. Let's keep using what we have and defer taking a new testing approach until after we have at least a few PRs on our journey of improving the recipes.

Reply:
This commit was pushed up in a state I had not intended to receive review. I had a 1-on-1 with @nealrichardson  earlier today and asked some questions about the approach to testing that I had taken with this PR, and he gave some thoughts on streamlining it that I implemented after our chat.

I was motivated to explore this approach as I was adding fixtures for the tests yesterday. There are three scenarios I want to test:

1. Both endpoints respond
2. The new endpoint returns a 404
3. The new and old endpoints returns a 404

So far, we have two mock dirs that are used for different versions of Connect. I can use those to mark out the two scenarios, but to test a third scenario requires adding another mock directory, and — I'm loathe to move to a model where we have additional mock dirs for each testing scenario, and managing tests of simple scenarios that are split out across an increasingly complex file tree has been a headache.

I threw this together in a few hours. It seems to work, and I @nealrichardson's suggestions made it much tighter. I also have found the workflow a good fit for me personally, and I think the tests are nicely concise and contained — it feels like a good fit for this kind of narrowly-scoped unit test.

I'm actually really excited about this approach — I feel like I've solved a piece of workflow friction that has been slowing me down.